examples/length_prefix_framing/chat-client.cpp

```diff
@@ -36,6 +38,7 @@ struct config : caf::actor_system_config {
   config() {
     opt_group{custom_options_, "global"} //
       .add<uint16_t>("port,p", "port of the server")
+      .add<uint16_t>("size,s", "length prefix size of the server")
```

Comment:
I like the idea of making this configurable. We shouldn't go through an integer though, since it requires manual conversions.

Move `size_field_type` to its own header under `caf/net/lp` and use our usual enum setup (declaring the three conversion functions plus `inspect` and then tell CMake to generate the scaffolding). You can look at `sec.hpp` for the setup (except not using `CAF_ERROR_CODE_ENUM`).

Once we have the proper setup, we can use the enum type directly here.